src/dataflow/components/dataflow-program.tsx

```diff
@@ -744,7 +751,9 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
       if (this.props.programEndTime && (Date.now() >= this.props.programEndTime)) {
         const hasDataStorage = this.getNodeCount("Data Storage") > 0;
         const programDisplayState = hasDataStorage ? ProgramDisplayStates.Graph : ProgramDisplayStates.Program;
-        this.setState({programRunState: ProgramRunStates.Complete, programDisplayState});
+        this.props.onCheckProgramRunState(this.props.programEndTime);
+        this.setState({ programRunState: ProgramRunStates.Complete, programDisplayState });
+        clearInterval(this.intervalHandle);
```

Comment:
I think that this `clearInterval` fixes a bug we saw during the team demo yesterday! 👍